machine_learning_hep/analysis/analyzerdhadrons_mult.py

```diff
@@ -62,10 +62,13 @@ def __init__(self, datap, case, typean, period):
                                   [None for _ in range(len(self.lvar2_binmin))])
         self.inel0_var = datap["analysis"][self.typean].get("inel0_var", "n_tracklets")
 
-        self.d_resultsallpmc = datap["analysis"][typean]["mc"]["results"][period] \
-                if period is not None else datap["analysis"][typean]["mc"]["resultsallp"]
-        self.d_resultsallpdata = datap["analysis"][typean]["data"]["results"][period] \
-                if period is not None else datap["analysis"][typean]["data"]["resultsallp"]
+        dp = datap["analysis"][typean]
+        self.d_prefix_mc = dp["mc"].get("prefix_dir_res")
+        self.d_prefix_data = dp["data"].get("prefix_dir_res")
+        self.d_resultsallpmc = self.d_prefix_mc + \
+            dp["mc"]["results"][period] if period is not None else dp["mc"]["resultsallp"]
+        self.d_resultsallpdata = self.d_prefix_data + \
+            dp["data"]["results"][period] if period is not None else dp["data"]["resultsallp"]
```

Comment:
More Pythonic way (avoid the explicit `\` for line continuation):
```suggestion
        self.d_resultsallpmc = (self.d_prefix_mc +
            dp["mc"]["results"][period] if period is not None else dp["mc"]["resultsallp"])
        self.d_resultsallpdata = (self.d_prefix_data +
            dp["data"]["results"][period] if period is not None else dp["data"]["resultsallp"])
```